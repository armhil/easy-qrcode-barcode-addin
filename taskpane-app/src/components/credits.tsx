/** credits route - acknowledging the dependencies */
import React from 'react';
import { LoggingUtils } from 'easy-addins-utils';
export function CreditsRoute() {
  // useEffect for tracing the route, should be logged only once.
  React.useEffect(() => {
    LoggingUtils.Trace('qrbar-credits');
  });

  return (
    <React.Fragment>
      <p>This software wouldn't be possible without components below and countless hours of learning and coding.</p>
      <ul>
        <li>
          <a href="https://developer.microsoft.com/en-us/fluentui"
          target="_blank" rel="noreferrer">Microsoft's Fluent UI</a></li>
        <li>
          <a href="https://github.com/zpao/qrcode.react"
          target="_blank" rel="noreferrer">Zpao's React QR Code Generator</a></li>
        <li>
          <a href="https://github.com/kciter/react-barcode"
          target="_blank" rel="noreferrer">Kciter's React Barcode Generator</a></li>
      </ul>
    </React.Fragment>
  )
}
