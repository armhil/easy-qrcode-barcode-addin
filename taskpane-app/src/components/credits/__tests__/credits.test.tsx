import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { LoggingUtils } from 'easy-addins-utils';
import { Credits } from '../credits';

describe('credits rendering', () => {
  it('should match with snapshot', () => {
    const tree = renderer.create(
      <Credits />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should log a trace', () => {
    const traceMockFn = jest.fn();
    LoggingUtils.Trace = traceMockFn;

    render(<Credits />);
    expect(traceMockFn).toHaveBeenCalledTimes(1);
    expect(traceMockFn).toHaveBeenCalledWith('qrbar-credits');
  })

  it('should render a links with correct attributes', () => {
    const dom = render(<Credits />);

    const links = dom.queryAllByRole('link');
    expect(links).not.toBeUndefined();
    // this is to ensure we can pop the users from Win32 apps
    links.forEach(t => {
      expect(t).toHaveAttribute('target', '_blank');
      expect(t).toHaveAttribute('rel', 'noreferrer');
    });
  });
})