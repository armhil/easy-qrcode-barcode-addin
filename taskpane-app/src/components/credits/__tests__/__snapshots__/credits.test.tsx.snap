// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`credits rendering should match with snapshot 1`] = `
[
  <p>
    This software wouldn't be possible without components below and countless hours of learning and coding.
  </p>,
  <ul>
    <li>
      <a
        href="https://developer.microsoft.com/en-us/fluentui"
        rel="noreferrer"
        target="_blank"
      >
        Microsoft's Fluent UI
      </a>
    </li>
    <li>
      <a
        href="https://github.com/zpao/qrcode.react"
        rel="noreferrer"
        target="_blank"
      >
        Zpao's React QR Code Generator
      </a>
    </li>
    <li>
      <a
        href="https://github.com/kciter/react-barcode"
        rel="noreferrer"
        target="_blank"
      >
        Kciter's React Barcode Generator
      </a>
    </li>
  </ul>,
]
`;
